lib/src/blaze/block_witness_data.rs

```diff
@@ -304,6 +265,58 @@ impl BlockAndWitnessData {
         return (true, heighest_tree);
     }
 
+    fn verify_tree_pair(
+        blocks: Arc<RwLock<Vec<BlockData>>>,
+        vt: TreeState,
+        ct: TreeState,
+    ) -> JoinHandle<bool> {
+        tokio::spawn(async move {
+            assert!(ct.height <= vt.height);
+
+            if ct.height == vt.height {
+                return true;
+            }
+            let mut sapling_tree =
+                CommitmentTree::<SaplingNode>::read(&hex::decode(ct.sapling_tree).unwrap()[..])
+                    .unwrap();
+            let mut orchard_tree = CommitmentTree::<MerkleHashOrchard>::read(
+                &hex::decode(ct.orchard_tree).unwrap()[..],
+            )
+            .unwrap_or(CommitmentTree::empty());
```

Comment:
It's interesting to me that we're being triggered by an `Result::Err` variant to supply a value (`CommitmentTree::empty()`).   This feels a bit odd to me, because it seems like a missing value is a more intuitive response to a `None` than an `Err`.

Reply:
Let's make this a match statement with explicit matches patterns so we know what event is triggering which arm.